### ISSUE TITLE
Investigate CI failure on release-1.3 branch

### ISSUE THREAD

**Issue by jpeach** [kumahq/kuma#2911]
### Summary

See https://app.circleci.com/pipelines/github/kumahq/kuma?branch=release-1.3

Investigate CI pipeline failures and make it green.

**Comment by jpeach** [comment]
> FYI: I'm attempting to fix the dev_{ubuntu,mac} failures over at: #2909 The flaky test/e2e ipv6/4 tests are fair game.

https://app.circleci.com/pipelines/github/kumahq/kuma/10314/workflows/4894455d-3138-4479-b339-c74c3cf33f76/jobs/114503

```
#!/bin/bash -eo pipefail
if [ -r /etc/os-release ]; then source /etc/os-release; fi
case "$ID" in
ubuntu)
  sudo apt update
  sudo env DEBIAN_FRONTEND=noninteractive apt install -y curl git make unzip gcc
  ;;
esac

/bin/bash: line 3: sudo: command not found

Exited with code exit status 127

```

What kind of ubuntu install doesn't have sudo?

**Comment by jpeach** [comment]
[This](https://app.circleci.com/pipelines/github/kumahq/kuma/10220/workflows/2971b117-17ba-4ecc-8726-171452431e78/jobs/113370) is the first release-1.3 branch pipeline run that failed to install k3d (see the `dev_mac` job).
This implicates #2862. For some reason, mergify ate the commit message on the backport, but the patch contents are the same.

Prior to that the `dev_mac` job failed for legitimate test reasons.
